subsys/debug/Kconfig

```diff
@@ -154,6 +154,18 @@ config UBSAN
 	  architecture, and requires a recent-ish compiler with the
 	  ``-fsanitize=undefined`` command line option.
 
+config MSAN
+	bool "Build with memory sanitizer"
+	depends on ARCH_POSIX
+	help
+	  Builds Zephyr with the MemorySanitizer enabled.  Works only
+	  on the native_posix architecutre currently, and only with
```

Comment:
Minor thing: the architecture would be "the POSIX architecture" native_posix is one of the boards using that architecture (another would be the nrf52_bsim).